src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -48,6 +50,19 @@ def get(self, request: Request, organization, project, monitor) -> Response:
         """
         Retrieves details for a monitor.
         """
+
+        environments = get_environments(request, organization)
+
+        if environments:
+            monitor_environment = MonitorEnvironment.objects.filter(
+                environment=environments[0], monitor=monitor
+            )
+            prefetch = Prefetch(
+                "monitorenvironment_set",
+                queryset=monitor_environment,
+                to_attr="selected_monitorenvironment",
+            )
+            monitor = Monitor.objects.prefetch_related(prefetch).get(id=monitor.id)
```

Comment:
don't like having to get the object out of the db again, but need it to work with the serializer for the list endpoint

Reply:
You can use https://docs.djangoproject.com/en/2.2/ref/models/querysets/#django.db.models.prefetch_related_objects to avoid this. Should be able to use `Prefetch` with this

Reply:
`prefetch_related` gives me the ability to set `attr` which I use for the serializer

Reply:
I thought that was part of the `Prefetch`? I haven't tested this tbh, but I assumed that `prefetch_related_object` would do the same thing.

Reply:
`Prefetch` is only used for `prefect_related`, not for `prefetch_related_objects`

Reply:
The docs I linked to say it'll accept a `Prefetch`. Seems to work

```
n [9]: pf  = Prefetch("monitorenvironment_set", to_attr="selected_monitorenvironment")

In [10]: monitor = Monitor.objects.first()

In [11]: monitor
Out[11]: <Monitor at 0x11e317730: id=1, guid=UUID('b7239e50-6b2d-45b6-a6c6-836553ada51f'), project_id=2, name='check-auth'>

In [15]: prefetch_related_objects([monitor], pf)

In [16]: monitor
Out[16]: <Monitor at 0x11e317730: id=1, guid=UUID('b7239e50-6b2d-45b6-a6c6-836553ada51f'), project_id=2, name='check-auth'>

In [17]: monitor.selected_monitorenvironment
Out[17]: []
```

I don't have any envs, but that seems to work ok?

Reply:
Could also just fetch the monitor and assign it, like `monitor.selected_monitorenvironment = me`, since we're only every fetching one

Reply:
for some reason i didn't think you couldn't assign arbitrary values like that...fixing. thanks